drivers/console/aarch64/multi_console.S

```diff
@@ -112,23 +112,23 @@ endfunc console_switch_state
 
 	/* -----------------------------------------------
 	 * void console_set_scope(console_t *console,
-	 *                       unsigned int scope)
+	 *                       u_register_t scope)
 	 * Function to update the states that a given console
 	 * may be active in.
 	 * In : x0 - pointer to console_t struct
-	 *    : w1 - new active state mask
+	 *    : x1 - new active state mask
 	 * Clobber list: x0, x1, x2
 	 * -----------------------------------------------
 	 */
 func console_set_scope
 #if ENABLE_ASSERTIONS
-	tst	w1, #~CONSOLE_FLAG_SCOPE_MASK
+	tst	x1, #~CONSOLE_FLAG_SCOPE_MASK
 	ASM_ASSERT(eq)
 #endif /* ENABLE_ASSERTIONS */
-	ldr	w2, [x0, #CONSOLE_T_FLAGS]
-	and	w2, w2, #~CONSOLE_FLAG_SCOPE_MASK
-	orr	w2, w2, w1
-	str	w2, [x0, #CONSOLE_T_FLAGS]
+	ldr	x2, [x0, #CONSOLE_T_FLAGS]
+	and	x2, x2, #~CONSOLE_FLAG_SCOPE_MASK
+	orr	x2, x2, x1
+	str	x2, [x0, #CONSOLE_T_FLAGS]
```

Comment:
We were writing 32 bits here, but we read 64 bits in the other functions. Check line 156, for example. This change was about making it consistent.

Reply:
In that case I'd say it would be better to change those other places to use 32-bit operations instead. My main issue with this patch is the unnecessary change of argument type to u_register_t.